src/test/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollectorTests.java

```diff
@@ -564,4 +564,64 @@ public void testTotalHitsCountValidation_whenTotalHitsCollectedAtTopLevelInColle
         reader.close();
         directory.close();
     }
+
+    @SneakyThrows
+    public void testMaxScoreValidation_whenMaxScoreCollectedAtTopLevelInCollector_thenSuccessful() {
```

Comment:
any possibility we can introduce multiple leaves in the test? The nature of the collector is to collect top results across leaves. And the tests should ideally reflect that behavior for it to be tight. See if you can have a case where you can have at least one more leaf

Reply:
The [collect](https://github.com/opensearch-project/neural-search/pull/765/files#diff-efec3f3ef0d15cf4737a062d0f2b229eb984463e066013395a9a5b7e1867a40eL94) method runs for every doc Id found as a result of the query on a particular shard. What you are expecting is a step which is way before this method to execute. 

Let me explain how this thing works

1) Initialize leaf collectors from all the collectors at a shard level. A leaf collector is unique entity.

For example
Let say there are 2 collectors TopDocCollector and HybridTopScoreDocCollector. These 2 collectors will have a unique leaf collectors of there own. Those leaf collectors will run once per shard. 

2) Then execute those leaf collectors (In this case https://github.com/opensearch-project/neural-search/pull/765/files#diff-efec3f3ef0d15cf4737a062d0f2b229eb984463e066013395a9a5b7e1867a40eR51)
This will like
leafCollector.collect (docId1) ---> this call will be made from OpenSearch
leafCollector.collect (docId2)
leafCollector.collect (docId3)

docId1, docId2, docId3 are the results of HybridQuery
3) return the result to coordinator node.

Reply:
While there is a new leafCollector, [you still are updating the priority queue and max score](https://github.com/opensearch-project/neural-search/pull/765/files#diff-efec3f3ef0d15cf4737a062d0f2b229eb984463e066013395a9a5b7e1867a40eL41) across collectors because its a member variable. And thats where the nature of this collector comes that is to collect the top docs across leaves and its important to test it out.